apps/web/components/pages/task/details-section/blocks/task-progress.tsx

```diff
@@ -5,74 +5,118 @@ import TaskRow from '../components/task-row';
+	hours: number;
+	minutes: number;
+}
 
 const TaskProgress = () => {
 	const [task] = useRecoilState(detailedTaskState);
-	const [dummyProfiles, setDummyProfiles] = useState<IEmployee[]>([]);
 	const { user } = useAuthenticateUser();
 	const { activeTeam } = useOrganizationTeams();
 	const { trans } = useTranslation('taskDetails');
 
-	//const seconds = useRecoilValue(timerSecondsState);
-	//const { activeTaskTotalStat, addSeconds } = useTaskStatistics(seconds);
+	const [userTotalTime, setUserTotalTime] = useState<ITime>({
+		hours: 0,
+		minutes: 0,
+	});
+	const [userTotalTimeToday, setUserTotalTimeToday] = useState<ITime>({
+		hours: 0,
+		minutes: 0,
+	});
+	const [timeRemaining, setTimeRemaining] = useState<ITime>({
+		hours: 0,
+		minutes: 0,
+	});
+	const [groupTotalTime, setGroupTotalTime] = useState<ITime>({
+		hours: 0,
+		minutes: 0,
+	});
+	const [numMembersToShow, setNumMembersToShow] = useState<number>(5);
 
 	const members = activeTeam?.members || [];
 
-	const currentUser = members.find((m) => {
+	const currentUser: OT_Member | undefined = members.find((m) => {
 		return m.employee.user?.id === user?.id;
 	});
 
-	const memberInfo = useTeamMemberCard(currentUser);
-
-	/*const TotalWork = () => {
-		if (memberInfo.isAuthUser) {
-			const { h, m } = secondsToTime(
-				//returns empty array
-				((currentUser?.totalTodayTasks &&
-					currentUser?.totalTodayTasks.reduce(
-						(previousValue, currentValue) =>
-							previousValue + currentValue.duration,
-						0
-					)) ||
-					activeTaskTotalStat?.duration ||
-					0) + addSeconds
-			);
-			return (
-
-					<div className="not-italic font-semibold text-xs leading-[140%] tracking-[-0.02em] text-[#282048] dark:text-white">
-						{h}h : {m}m
-					</div>
+	// const memberInfo = useTeamMemberCard(currentUser);
+
+	useEffect(() => {
+		const userTotalTimeOnTask = (): void => {
+			const totalOnTaskInSeconds: number =
+				currentUser?.totalWorkedTasks.find((object) => object.id === task?.id)
+					?.duration || 0;
+
+			const { h, m } = secondsToTime(totalOnTaskInSeconds);
 
-			);
-		}
-	};*/
+			setUserTotalTime({ hours: h, minutes: m });
+		};
+		userTotalTimeOnTask();
+	}, [currentUser?.totalWorkedTasks, task?.id]);
 
 	useEffect(() => {
-		if (task && task?.members) {
-			const profiles = Array.isArray(task?.members) ? [...task.members] : [];
+		const userTotalTimeOnTaskToday = (): void => {
+			const totalOnTaskInSeconds: number =
+				currentUser?.totalTodayTasks.find((object) => object.id === task?.id)
+					?.duration || 0;
+
+			const { h, m } = secondsToTime(totalOnTaskInSeconds);
+
+			setUserTotalTimeToday({ hours: h, minutes: m });
+		};
+		userTotalTimeOnTaskToday();
+	}, [currentUser?.totalTodayTasks, task?.id]);
+
+	useEffect(() => {
+		const matchingMembers: OT_Member[] | undefined = activeTeam?.members.filter(
+			(member) =>
+				task?.members.some((taskMember) => taskMember.id === member.employeeId)
+		);
+		console.log('matchingMembers:', matchingMembers);
```

Comment:
@desperado1802 let's remove this console, and any other consoles that we by mistake forgot to remove!